src/lib/DefaultDependencyManager.ts

```diff
@@ -0,0 +1,41 @@
+import {Logger, SfdxError} from '@salesforce/core';
+import {Controller} from '../Controller';
+import {OUTPUT_FORMAT} from '../Constants';
+import {DependencyChecker} from './services/DependencyChecker';
+import {DependencyResult} from '../types';
+import {DependencyManager} from './DependencyManager';
+
+export class DefaultDependencyManager implements DependencyManager {
+	private logger: Logger;
+	private initialized: boolean;
+
+
+	async init(): Promise<void> {
+		if (this.initialized) {
+			return;
+		}
+		this.logger = await Logger.child('DefaultDependencyManager');
+		this.initialized = true;
+	}
+
+	public async scanForInsecureDependencies(target: string, format: OUTPUT_FORMAT): Promise<any> {
+		const resultPromises: Promise<DependencyResult[]>[] = [];
+
+		const depCheckers: DependencyChecker[] = await Controller.getEnabledDepCheckers();
+		for (const d of depCheckers) {
+			// Spin up a Promise for each dependency checker.
+			this.logger.trace(`${d.getName()} is eligible to execute`);
+			// TODO: As we add more functionality, there will definitely need to be changes here.
+			resultPromises.push(d.run(target));
+		}
+
+		try {
+			// TODO: As we add more functionality and flesh out our design, there will definitely need to be changes here.
+			const results = (await Promise.all(resultPromises)).reduce((acc, v) => [...acc, ...v], []);
```

Comment:
What are `acc` and `v`? Could they be renamed to be more descriptive, or a comment added?

Reply:
`acc` is "accumulator" and `v` is just "value". They're pretty standard terms for the `reduce` method, which is easier to understand with examples than with an explanation.
This:
`let arr = [5, 6, 7, 8];`
`let x = arr.reduce((acc, v) => acc + v, 0);`
is equivalent to this:
`let arr = [5, 6, 7, 8];`
`let x = 0;`
`for (const v of arr) {`
`x += v;`
`}`

Reply:
Thanks, a comment like this one would help the reader. https://github.com/forcedotcom/sfdx-scanner/blob/dev/src/lib/DefaultRuleManager.ts#L78

For the sake of consistency, it would help to change DefaultRuleManager to the `reduce` pattern or change this code to the `forEach` patterrn.